pkg/discovery/config.go

```diff
@@ -55,16 +55,25 @@ func (c Config) Run(ctx context.Context, server *provider.Provider) (err error)
 
 	discovery := NewDiscovery(zap.L().Named("discovery"), ol, kad, stat.StatDB())
 
-	zap.L().Debug("Starting discovery")
+	zap.L().Debug("bootstrapping cache")
+	err = discovery.Bootstrap(ctx)
+	if err != nil {
+		discovery.log.Error("Error bootstrapping cache", zap.Error(err))
+	}
 
+	zap.L().Debug("Starting discovery")
 	ticker := time.NewTicker(c.RefreshInterval)
 	defer ticker.Stop()
 
 	go func() {
 		for {
 			select {
 			case <-ticker.C:
-				err := discovery.Refresh(ctx)
+				err := discovery.Bootstrap(ctx)
```

Comment:
Have we changed the meaning of bootstrap by calling it regularly?  Do we always call bootstrap now before a refresh?  Should we move the call to Bootstrap to within Refresh to avoid naming confusion?

Reply:
I fixed this with that context bug fix I mentioned.